packages/astro/src/content/runtime.ts

```diff
@@ -624,8 +617,9 @@ export function createReference({ lookupMap }: { lookupMap: ContentLookupMap })
 						return { id: lookup, collection };
 					}
 
-					if (!lookupMap[collection] && store.collections().size === 0) {
+					if (!lookupMap[collection] && store.collections().size < 2) {
```

Comment:
Now that we have a "magic number", I think it's worth adding a comment that explains this check and why "2", so we don't lose this valuable knowledge

Reply:
The comment is just below, but I'll add one that's more explicit above